src/internal/pjs/server_internal_test.go

```diff
@@ -301,14 +324,14 @@ func TestWalkJob(t *testing.T) {
 
 // runJob does work through PJS.
 func runJob(t *testing.T, ctx context.Context, c pjs.APIClient, fc storage.FilesetClient, in *pjs.CreateJobRequest,
-	fn func(resp *pjs.ProcessQueueResponse) error) (*pjs.JobInfo_Success, error) {
+	fn func(resp *pjs.ProcessQueueResponse) error) (*pjs.JobInfo, error) {
 	jres, err := c.CreateJob(ctx, in)
 	require.NoError(t, err)
 	return runJobFrom(t, ctx, c, fc, jres.Id.Id, in.Program, fn)
 }
 
 func runJobFrom(t *testing.T, ctx context.Context, c pjs.APIClient, fc storage.FilesetClient, from int64, programStr string,
-	fn func(resp *pjs.ProcessQueueResponse) error) (*pjs.JobInfo_Success, error) {
+	fn func(resp *pjs.ProcessQueueResponse) error) (*pjs.JobInfo, error) {
```

Comment:
Rather than returning a full Job Info message, would it make sense to declare a  Job Info Result submessage and use that instead?

Reply:
Makes sense. I thought about this, but I'm not sure which the best way is to do so.
This is how result is defined in JobInfo:
`	// Types that are assignable to Result:
	//
	//	*JobInfo_Success_
	//	*JobInfo_Error
	Result isJobInfo_Result `protobuf_oneof:"result"``

The isJobInfo_Result is a generated struct in pjs.pb.go and it's not exported. So should I create another type or struct that somehow wraps around them?

Reply:
> So should I create another type or struct that somehow wraps around them?

Since these functions are just used in testing, I would probably just declare a local unexported type, e.g.:

```go
type jobInfoResult struct {
  success *pjs.JobInfo_Success_
  error *pjs.JobInfo_Error
}
```

And then just take it apart accordingly in your testing code.

If you were going to use these functions in the actual package, I would probably create a submessage:

```protobuf
  message Result {
    // result is set when the Job enters the DONE state.
    oneof result {
      Success success = 1;
      // error is set when the Job is unable to complete successfully
      JobErrorCode error = 2;
    }
  }
  Result result = 6;
```